### PR TITLE
Replace std::ostringstream with std::to_string (3) Issue #388

### LINKED ISSUE #388: Replace std::ostringstream with std::to_string
This is another easy task to fix that I have seen a lot: there are lots of places where we write, e.g.,

```cpp
std::ostringstream bc_coefs_name_stream;
bc_coefs_name_stream << "u_bc_coefs_" << d;
const std::string bc_coefs_name = bc_coefs_name_stream.str();
```
when, with C++11, we could simply write
```cpp
const std::string bc_coefs_name = "u_bc_coefs_" + std::to_string(d);
```

### PR DESCRIPTION
**Pull request by jabrown893**
This pull request resolves #388 by addressing occurrences of the issue in the examples, tests, and ibtk/examples folders.

### CODE REVIEW

**Comment by boyceg** [comment]
👍

File: examples/ConstraintIB/flow_past_cylinder_HF/HydroForceEval.cpp [hunk 1]
```diff
@@ -55,9 +55,7 @@ template <typename T>
 std::string
 to_string(const T& n)
 {
-    std::ostringstream stm;
-    stm << n;
-    return stm.str();
+    return std::to_string(n);
 }
```
Comment (drwells) [inline examples/ConstraintIB/flow_past_cylinder_HF/HydroForceEval.cpp]: We can probably just get rid of this function altogether.

File: examples/ConstraintIB/flow_past_cylinder_HF/HydroForceEval.cpp [hunk 2]
```diff
@@ -132,12 +130,10 @@ HydroForceEval::HydroForceEval(const std::string& object_name,
...
         for (int struct_no = 0; struct_no < d_num_structs; ++struct_no)
         {
-            std::ostringstream force_stream;
-            force_stream << d_dir_name + '/' + d_struct_names[struct_no] + "_HF.txt";
             if (from_restart)
-                d_force_stream[struct_no] = new std::ofstream(force_stream.str().c_str(), std::fstream::app);
+                d_force_stream[struct_no] = new std::ofstream((d_dir_name + '/' + d_struct_names[struct_no] + "_HF.txt").c_str(), std::fstream::app);
             else
-                d_force_stream[struct_no] = new std::ofstream(force_stream.str().c_str(), std::fstream::out);
+                d_force_stream[struct_no] = new std::ofstream((d_dir_name + '/' + d_struct_names[struct_no] + "_HF.txt").c_str(), std::fstream::out);
```
Comment (drwells) [inline examples/ConstraintIB/flow_past_cylinder_HF/HydroForceEval.cpp]: While we are here: the calls to `c_str()` are no longer necessary.

File: examples/multiphase_flow/ex0/example.cpp [hunk 3]
```diff
@@ -500,4 +496,4 @@ output_data(Pointer<PatchHierarchy<NDIM> > patch_hierarchy,
     hier_db->putInteger("iteration_num", iteration_num);
     hier_db->close();
     return;
-} // output_data
\ No newline at end of file
+} // output_data
```
Comment (drwells) [inline examples/multiphase_flow/ex0/example.cpp]: :+1:

File: examples/ConstraintIB/flow_past_cylinder_HF/HydroForceEval.cpp (hunk 1 above)
Comment (jabrown893) [reply examples/ConstraintIB/flow_past_cylinder_HF/HydroForceEval.cpp]: @drwells , so, delete lines 55 through 59?

File: examples/CIB/ex0/example.cpp [hunk 4]
```diff
@@ -236,13 +236,9 @@ run_example(int argc, char* argv[])
...
             {
-                ostringstream bc_coefs_name_stream;
-                bc_coefs_name_stream << "u_bc_coefs_" << d;
-                const string bc_coefs_name = bc_coefs_name_stream.str();
+                const string bc_coefs_name = "u_bc_coefs_" + std::to_string(d);
 
-                ostringstream bc_coefs_db_name_stream;
-                bc_coefs_db_name_stream << "VelocityBcCoefs_" << d;
-                const string bc_coefs_db_name = bc_coefs_db_name_stream.str();
+                const string bc_coefs_db_name = "VelocityBcCoefs_" + std::to_string(d);
```
[the remark this answers is not in the document]
Comment (jabrown893) [reply examples/CIB/ex0/example.cpp]: Ok, I think I have fixed all occurrences of this.

File: examples/ConstraintIB/flow_past_cylinder_HF/HydroForceEval.cpp (hunk 2 above)
Comment (jabrown893) [reply examples/ConstraintIB/flow_past_cylinder_HF/HydroForceEval.cpp]: Ok, got it.

File: examples/ConstraintIB/flow_past_cylinder_HF/HydroForceEval.cpp (hunk 1 above)
Comment (drwells) [reply examples/ConstraintIB/flow_past_cylinder_HF/HydroForceEval.cpp]: Yes. We should be able to just use `std::to_string` everywhere this was used.

Comment (jabrown893) [reply examples/ConstraintIB/flow_past_cylinder_HF/HydroForceEval.cpp]: @drwells Ok, I think it's good now.

File: examples/ConstraintIB/flow_past_cylinder_HF/HydroForceEval.cpp [hunk 5]
```diff
@@ -124,20 +112,18 @@ HydroForceEval::HydroForceEval(const std::string& object_name,
         Utilities::recursiveMkdir(d_dir_name);
 
         const int nodes = SAMRAI_MPI::getNodes();
-        for (int n = 0; n < nodes; ++n) Utilities::recursiveMkdir(d_dir_name + "/node" + patch::to_string(n));
+        for (int n = 0; n < nodes; ++n) Utilities::recursiveMkdir(d_dir_name + "/node" + std::to_string(n));
```
Comment (drwells) [inline examples/ConstraintIB/flow_past_cylinder_HF/HydroForceEval.cpp]: :heavy_check_mark:

**Pull request merged**